src/components/Api.js

```diff
@@ -32,41 +25,44 @@ class Api {
       .catch(this._handleError);
   }
 
-  setUserInfo(data) {
-    return fetch(`${this._baseUrl}/users/me`, {
-      method: "PATCH",
+  // 2. Loading cards from the server
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
       headers: this._headers,
-      body: JSON.stringify({
-        name: data.name,
-        about: data.about,
-      }),
     })
       .then(this._checkResponse)
       .catch(this._handleError);
   }
 
-  setUserAvatar(data) {
-    return fetch(`${this._baseUrl}/users/me/avatar`, {
+  // 3. Editing the profile
+  setUserInfo(data) {
+    return fetch(`${this._baseUrl}/users/me`, {
       method: "PATCH",
-      headers: this._headers,
-      body: JSON.stringify({
-        avatar: data.avatar,
-      }),
+      headers: {
+        ...this._headers,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
     })
       .then(this._checkResponse)
       .catch(this._handleError);
```

Comment:
Needs correction: Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```